datadog_checks_downloader/tests/scripts/download_test_data.py

```diff
@@ -0,0 +1,137 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+"""
+Script to update test data.
+
+It downloads a minimal subset of files from the actual online repository which is
+enough to test downloading an integration. The files are saved in zip files under
+tests/data.
+"""
+
+import json
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from urllib.parse import urljoin
+from urllib.request import urlopen
+from zipfile import ZipFile
+
+REPOSITORY_BASE_URL = 'https://dd-integrations-core-wheels-build-stable.datadoghq.com/'
+INTEGRATION = 'active-directory'
+INTEGRATION_VERSION = '1.10.0'
+GOOD_REPO_ZIP_FILENAME = f'datadog-{INTEGRATION}-{INTEGRATION_VERSION}.zip'
+BAD_REPO_ZIP_FILENAME = f'datadog-{INTEGRATION}-{INTEGRATION_VERSION}-signature-wheels-signer-a.zip'
+TARGET_DIR = Path(__file__).parent.parent / 'data'
+
+
+def main():
+    with TemporaryDirectory() as tempdir:
+        tempdir = Path(tempdir)
+
+        files_to_zip = set()
+        versioned_metadata_files = {}
+        hashed_target_files = {}
+
+        def download_file(relative_url):
```

Comment:
Would it make sense to remove nested functions if they are not necessary? They might make sources harder to read (also test, but that's probably not the case here).

Reply:
They're nested so that they can access the common context / state and be more convenient to use. I don't think it means much in this particular case because I could remove the `main` function and have global instead of function-local variables and functions.

Reply:
It's maybe nitpicking from my side - imho this script does not need a lot of engineering time investment.

BTW, for similar cases (when a state is shared), a class could be considered.